modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt

```diff
@@ -400,18 +405,26 @@ internal class DiscoverAdapter(
                 }
                 is CarouselListViewHolder -> {
                     val featuredLimit = 5
+
                     val loadingFlowable: Flowable<List<Any>> = loadPodcastList(row.source)
-                        .flatMap<List<Any>> { podcastList ->
-                            Flowable.fromIterable(podcastList.podcasts)
-                                .take(featuredLimit.toLong())
-                                .concatMap {
+                        .zipWith(loadCarouselSponsoredPodcastList(row.sponsoredPodcasts))
+                        .flatMap {
+                            val (featuredPodcastList, sponsoredPodcastList) = it
+                            val mutableList = featuredPodcastList.podcasts
+                                .take(featuredLimit)
+                                .toMutableList()
+                            sponsoredPodcastList.forEach { sponsoredPodcast ->
+                                mutableList.addSafely(sponsoredPodcast.podcast.copy(isSponsored = true), sponsoredPodcast.position)
```

Comment:
I wasn't expecting that we grow the size of the carousel with each sponsored podcast (I thought the carousel size would remain the same regardless of the number of sponsored podcasts). What you have also makes sense, I just wanted to confirm that it is the desired behavior.

Reply:
I'll keep the existing logic to be consistent with iOS.